lib/src/facts/openbsd/memory_resolver.cc

```diff
@@ -37,6 +37,38 @@ namespace facter { namespace facts { namespace openbsd {
             result.mem_free = static_cast<u_int64_t>(uvmexp.free) << uvmexp.pageshift;
         }
 
+        struct swapent *swapdev;
+        int nswaps, s;
+        uint64_t swap_used = 0;
+
+        // NB: swapctl(2) for SWAP_NSWAP cannot fail
+        nswaps = swapctl(SWAP_NSWAP, 0, 0);
+
+        if ((swapdev = (struct swapent *)calloc(nswaps, sizeof(*swapdev))) == NULL)
```

Comment:
This seems to be leaked memory. To avoid this, we've tended to use [RAII](https://en.wikipedia.org/wiki/Resource_Acquisition_Is_Initialization), as in https://github.com/puppetlabs/facter/blob/master/lib/inc/internal/util/posix/scoped_addrinfo.hpp.